### ISSUE TITLE
[Bug]: Search often times out, which takes several minutes

### ISSUE THREAD

**Issue by yesletgo** [milvus-io/milvus#22799]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.1.4
- Deployment mode(standalone or cluster):cluster
- MQ type(rocksmq, pulsar or kafka):    pulsar 
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): CentOS
- CPU/Memory: 96cores/682.59 GiB
- GPU: 
- Others:
```

### Current Behavior

**The amount of data at the level of 100 million will occasionally take a few minutes to search the topk 50, and the rest will basically return within 100 ms.**

**I'm experiencing**
### 
2023-03-16 14:20:42,874 [971d4003] [INFO  ] [http-nio-8081-exec-5] AbstractMilvusGrpcClient.java:2056 - SearchParam{collectionName='vtl_v4', partitionNames='[]', metricType=IP, target vectors count=1, vectorFieldName='feature', topK=50, expr='', params='{"nprobe":1}', consistencyLevel='EVENTUALLY'}
2023-03-16 14:20:53,379 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 50  actions to finish on table: vtl
2023-03-16 14:21:03,380 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 50  actions to finish on table: vtl
2023-03-16 14:21:13,389 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 50  actions to finish on table: vtl
2023-03-16 14:21:23,398 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 50  actions to finish on table: vtl
2023-03-16 14:21:33,405 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 50  actions to finish on table: vtl
2023-03-16 14:21:43,408 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 50  actions to finish on table: vtl
2023-03-16 14:21:53,414 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:22:03,422 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:22:13,425 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:22:23,431 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:22:33,436 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:22:43,440 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:22:53,447 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:23:03,454 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:23:13,462 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:23:23,469 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:23:33,475 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:23:43,484 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:23:53,490 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:24:03,500 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:24:13,507 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:24:23,517 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:24:33,525 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
2023-03-16 14:24:43,529 [971d4003] [INFO  ] [http-nio-8081-exec-5] AsyncRequestFutureImpl.java:1175 - #1, waiting for 26  actions to finish on table: vtl
### 

### 
2023-03-16 14:39:31,106 [726250d6] [INFO  ] [http-nio-8081-exec-3] AbstractMilvusGrpcClient.java:2056 - SearchParam{collectionName='vtl_v4', partitionNames='[]', metricType=IP, target vectors count=1, vectorFieldName='feature', topK=50, expr='', params='{"nprobe":1}', consistencyLevel='EVENTUALLY'}
2023-03-16 14:39:41,617 [726250d6] [INFO  ] [http-nio-8081-exec-3] AsyncRequestFutureImpl.java:1175 - #1, waiting for 39  actions to finish on table: vtl
2023-03-16 14:39:51,623 [726250d6] [INFO  ] [http-nio-8081-exec-3] AsyncRequestFutureImpl.java:1175 - #1, waiting for 39  actions to finish on table: vtl
2023-03-16 14:40:01,633 [726250d6] [INFO  ] [http-nio-8081-exec-3] AsyncRequestFutureImpl.java:1175 - #1, waiting for 39  actions to finish on table: vtl
2023-03-16 14:40:11,641 [726250d6] [INFO  ] [http-nio-8081-exec-3] AsyncRequestFutureImpl.java:1175 - #1, waiting for 39  actions to finish on table: vtl
2023-03-16 14:40:21,648 [726250d6] [INFO  ] [http-nio-8081-exec-3] AsyncRequestFutureImpl.java:1175 - #1, waiting for 39  actions to finish on table: vtl
2023-03-16 14:40:31,656 [726250d6] [INFO  ] [http-nio-8081-exec-3] AsyncRequestFutureImpl.java:1175 - #1, waiting for 39  actions to finish on table: vtl

### Expected Behavior

**Hope to return in 100ms and reduce the number of search timeout**

### Milvus Log

1. [milvus-log.tar.gz](https://github.com/milvus-io/milvus/files/10989245/milvus-log.tar.gz)
2. collection name: vtl_v4
3. search time:  **2023-03-16 14:20:42,874** and **2023-03-16 14:39:31,106**

**Comment by xiaofan-luan** [comment]
> 
@yesletgo

is this a production environment?Could you try to upgrade to 2.2.3 see if the problem still exist

**Comment by yesletgo** [comment]
> > 
> 
> @yesletgo
> 
> is this a production environment?Could you try to upgrade to 2.2.3 see if the problem still exist

It's the production environment, so we want to find out the cause of the problem. We will upgrade to the latest version in the future

> > 
> 
> @yesletgo
> 
> is this a production environment?Could you try to upgrade to 2.2.3 see if the problem still exist

It's the production environment, so we want to find out the cause of the problem. We will upgrade to the latest version in the future.

**Comment by yesletgo** [comment]
With sufficient resources, the following two situations occasionally occur:

1: Unable to obtain index information from metadata

2: Flush timeout

From code:
initBuildIDs, err := c.MetaTable.GetInitBuildIDs(in.GetCollectionName(), in.GetIndexName())
resp, err := cli.FlushSegments(ctx, req)

Can you help me explain this phenomenon from the bottom？

**Comment by yesletgo** [comment]
Ok. There is no manual GetInitBuildID and flush. This is what I found out by following the source code and the error log. Searching meta_table metadata and flush may be two very important links. If it is unstable, it will have a great impact.

**Comment by xiaofan-luan** [comment]
> Ok. There is no manual GetInitBuildID and flush. This is what I found out by following the source code and the error log. Searching meta_table metadata and flush may be two very important links. If it is unstable, it will have a great impact.

I just quickly check the logs, the flush is very frequent and seems to be known issue for 2.1.X. Upgrade to 2.2.3/2.2.4 should automatically solve the issue because the code is very different from 2.1.X to now and we won't release any fix on 2.1

**Comment by yanliang567** [comment]
close for now as no updates for long time
